rules.py

```diff
@@ -1010,6 +1010,52 @@ def check(self, element):
                             element.sourceline, element.tag, other_type_element.text))
 
 
+class GpUnitsCycleCheck(base.TreeRule):
```

Comment:
This check doesn't work as advertised. For example, if the same precinct appears in multiple high-level units (e.g., it appears in one state senate district and one state house district) this code will flag it as creating a cycle, when in fact it is not.

Please test code against the included files before creating pull requests:
post_election_sample_feed_precincts.xml
post_election_sample_feed_summary.xml
pre_election_sample_feed.xml